pyneuroml/plot/Plot.py

```diff
@@ -240,6 +249,41 @@ def generate_plot(
         logger.info("Saved image to %s of plot: %s" % (save_figure_to, title))
 
     if show_plot_already:
+        if animate:
+            d = 5
+            duration = d * 1000 # in ms
+            size = len(xvalues[0])
+            interval = 50  # Delay between frames in milliseconds
+            pockets = duration // interval
+            skip = max(size // pockets, 1)
+            logger.info(
+                "Animation hyperparameters : duration=%s, size=%s, interval=%s, pockets=%s, skip=%s" % (duration, size, interval, pockets,  skip))
+
+            def update(frame):
+                for i, artist in enumerate(artists):
+                    artist.set_xdata(xvalues[i][:frame*skip])
+                    artist.set_ydata(yvalues[i][:frame*skip])
+                return artists
+            
+            ani = animation.FuncAnimation(
+                fig=fig,
+                frames=size-1,
+                func=update,
+                interval=interval,
+                blit=True,
+                cache_frame_data=False
+            )
+
+            frame_length_threshold = 5000
+            if len(xvalues[0]) < frame_length_threshold and save_animation_to:
+                logger.info("Saving animation to %s" %
+                            (save_animation_to))
+                ani.save(
+                    filename=save_animation_to,
+                    writer="pillow",
```

Comment:
I wonder if we should make this an argument too, to allow the user to choose different writers? ffmpeg/imagemagick may be quicker than pillow and support more formats:

https://matplotlib.org/stable/users/explain/animations/animations.html

How about making it an argument of the form:

```
writer={ 'writer name': ["extra args list"] }
```

with default value:

```
writer = { "pillow": [] }
```

?

Reply:
We should also add this link to the docstring around these arguments:

https://matplotlib.org/stable/users/explain/animations/animations.html#saving-animations

Reply:
should I include this in the same PR or should it be done in another PR.

Reply:
I think we can roll this in here too?